File: src/main/java/com/zwk/node/RootNode.java
package com.zwk.node;

import java.util.ArrayList;
import java.util.List;

/**
 * @author zwk
 * @version 1.0
 * @date 2024/4/3 14:50
 */

public class RootNode extends BaseNode {

    public List<Node> imports = new ArrayList<>();

    public List<Node> inits = new ArrayList<>();
    public List<Node> nodes = new ArrayList<>();


    @Override
    public <R, C> R accept(NodeVisitor<R, C> visitor, C c) {
        return visitor.visitRoot(this, c);
    }
}

File: src/main/kotlin/com/zwk/node/RootNode.kt
package com.zwk.node;

/**
 * @author zwk
 * @version 1.0
 * @date 2024/4/3 14:50
 */

class RootNode : BaseNode() {

    val imports = mutableListOf<Node>()
    val inits = mutableListOf<Node>()
    val funs = mutableListOf<Node>()

    override fun <R, C> accept(visitor: NodeVisitor<R, C>, c: C): R {
        return visitor.visitRoot(this, c)
    }
}
